fix(ci): install ImageMagick on runner in social-post workflow

## Problem

The `social-post.yml` workflow fails at the Instagram crop step with:

```
convert: command not found
Error: Process completed with exit code 127.
```

ImageMagick is not pre-installed on `ubuntu-latest` GitHub Actions runners.

## Fix

Add an install step before the Generate Instagram crop step:

\`\`\`yaml
- name: Install ImageMagick
  run: sudo apt-get install -y imagemagick
\`\`\`

## Discovered in

Issue #222 (first real run of the social-post backfill workflow).

--- a/.github/workflows/social-post.yml
+++ b/.github/workflows/social-post.yml
@@ -113,6 +113,9 @@ jobs:
           print(f'Hero image: {hero_image}')
           PYEOF
 
+      - name: Install ImageMagick
+        run: sudo apt-get install -y imagemagick
+
       - name: Generate Instagram crop
         id: crop
         env:

